lib/matplotlib/axes/_base.py

```diff
@@ -2673,10 +2673,13 @@ def set_xscale(self, value, **kwargs):
         # around zero
         if value.lower() == 'log' and 'nonposx' not in kwargs.keys():
             kwargs['nonposx'] = 'clip'
-        self.xaxis._set_scale(value, **kwargs)
-        self.autoscale_view(scaley=False)
-        self._update_transScale()
-        self.stale = True
+
+        g = self.get_shared_x_axes()
+        for ax in g.get_siblings(self):
+            ax.xaxis._set_scale(value, **kwargs)
+            ax.autoscale_view(scaley=False)
```

Comment:
Yup, just checked:

```
        if scalex and self._autoscaleXon:
            xshared = self._shared_x_axes.get_siblings(self)
            dl = [ax.dataLim for ax in xshared]
            # ignore non-finite data limits if good limits exist
            finite_dl = [d for d in dl if np.isfinite(d).all()]
            if len(finite_dl):
                dl = finite_dl
```
So, with this design, there is a potential for a huge performance hit, even for moderately sized plot grids.